Added more JSON formats

andrewzolotukhin opened the pull request:
## Summary
Added more JSON formats. And fixed issue with 'JSONObjectEachRow' throwing an error.
Added support for the following formats: JSONStrings, JSONCompact, JSONCompactStrings, JSONColumnsWithMetadata.

## Checklist
Delete items not relevant to your PR:
- [ ] A human-readable description of the changes was provided to include in CHANGELOG

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ClickHouse/clickhouse-js?pullRequest=142) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ClickHouse/clickhouse-js?pullRequest=142) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ClickHouse/clickhouse-js?pullRequest=142) it.</sub>

slvrtrn commented on src/data_formatter/formatter.ts:
```diff
@@ -8,9 +8,16 @@ const streamableJSONFormats = [
   'JSONObjectEachRow',
+] as const
+const supportedJSONFormats = [
+  ...notStreamableJSONFormats,
```
nit: `nonStreamableJSONFormats` (non as a prefix)

slvrtrn commented on src/data_formatter/formatter.ts:
```diff
@@ -8,9 +8,16 @@ const streamableJSONFormats = [
+  'JSONStrings',
+  'JSONCompact',
+  'JSONCompactStrings',
+  'JSONColumnsWithMetadata',
```
Could you add these new formats to `encode_values.test.ts`?

slvrtrn reviewed (commented):
Thanks for the contribution! Let's add a few tests here, and we can merge this.

andrewzolotukhin commented:
Hi! I've addressed your comments. Do you have any kind of release schedule? Just wondering when these changes are going to be published in npm (I going to use it in my project).

Thank You!

slvrtrn commented:
@andrewzolotukhin I will release it ASAP (maybe even today).
Please stay tuned.

Again, thanks for your contribution.

slvrtrn commented:
@andrewzolotukhin 0.0.14 is out, including this changeset.
